datadog/resource_datadog_dashboard_list.go

```diff
@@ -80,61 +84,51 @@ func resourceDatadogDashboardListCreate(d *schema.ResourceData, meta interface{}
 
 func resourceDatadogDashboardListUpdate(d *schema.ResourceData, meta interface{}) error {
 	providerConf := meta.(*ProviderConfiguration)
-	client := providerConf.CommunityClient
+	datadogClientV1 := providerConf.DatadogClientV1
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV1 := providerConf.AuthV1
+	authV2 := providerConf.AuthV2
 
-	id, err := strconv.Atoi(d.Id())
+	id, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return fmt.Errorf("failed to parse resource id: %s", err.Error())
+	}
 
 	// Make any necessary updates to the Overall Dashboard List Object
 	dashList, err := buildDatadogDashboardList(d)
 	if err != nil {
 		return fmt.Errorf("failed to parse resource configuration: %s", err.Error())
 	}
-	dashList.SetId(id)
 	dashList.SetName(d.Get("name").(string))
-	err = client.UpdateDashboardList(dashList)
+	_, _, err = datadogClientV1.DashboardListsApi.UpdateDashboardList(authV1, id).Body(dashList).Execute()
 	if err != nil {
 		return translateClientError(err, "error updating dashboard list")
 	}
 
-	// Delete all elements from the dash list and add back only the ones in the config
-	completeDashListV2, err := client.GetDashboardListItemsV2(id)
-	if err != nil {
-		return translateClientError(err, "error getting dashboard list item")
-	}
-	completeDashListV2, err = client.DeleteDashboardListItemsV2(id, completeDashListV2)
-	if err != nil {
-		return translateClientError(err, "error deleting dashboard list item")
-	}
-	if len(d.Get("dash_item").(*schema.Set).List()) > 0 {
-		dashboardListV2Items, err := buildDatadogDashboardListItemsV2(d)
-		if err != nil {
-			return fmt.Errorf("failed to parse resource configuration: %s", err.Error())
-		}
-		_, err = client.UpdateDashboardListItemsV2(id, dashboardListV2Items)
-		if err != nil {
-			return translateClientError(err, "error updating dashboard list item")
-		}
-	}
-
+	// Update the dashboard list
```

Comment:
Why is this different from what is above?

Reply:
I will sync with Nick on this one to see why it was previously doing the extra step of both deleting the list items and manually adding the items in the config. The api endpoint receives the updated Dash and replaces the previous list. Since the api does this automatically, we were doing extra unnecessary steps. 

But I will follow up here after discussing with Nick

Reply:
I think we should keep the old logic. However, I am not sure why we do `client.UpdateDashboardListItemsV2(id, dashboardListV2Items)` when in fact we should create them.